cgi/product_jqm_multilingual.pl

```diff
@@ -59,11 +59,38 @@
 
 my %response = ();
 
-my $code = normalize_code(param('code'));
+my $code = param('code');
 my $product_id;
 
 $log->debug("start", { code => $code, lc => $lc }) if $log->is_debug();
 
+# Allow apps to create products without barcodes
+# Assign a code and return it in the response.
+if ($code eq "new") {
+
+	$code = 2000000000001; # Codes beginning with 2 are for internal use
+
+	my $internal_code_ref = retrieve("$data_root/products/internal_code.sto");
+	if ((defined $internal_code_ref) and ($$internal_code_ref > $code)) {
+		$code = $$internal_code_ref;
+	}
+
+	$product_id = product_id_for_user($User_id, $Org_id, $code);
+
+	while (-e ("$data_root/products/" . product_path_from_id($product_id))) {
+
+		$code++;
+		$product_id = product_id_for_user($User_id, $Org_id, $code);
+	}
+
+	store("$data_root/products/internal_code.sto", \$code);
```

Comment:
This seems to be the same as https://github.com/openfoodfacts/openfoodfacts-server/blob/0d871ca02bfe9aa93bcd6816ff85bb5aa090fd0f/cgi/product_multilingual.pl#L90-L106. Wouldn't it make sense to refactor it to a method in `ProductOpener::Products`?

Reply:
@hangy : I just refactored it, thanks!